### PR TITLE
Support DNSSEC commands in the CLI

### PR DESCRIPTION
**Pull request by grigouze**

### CODE REVIEW

File: gandicli.man.rst [hunk 1]
```diff
@@ -98,6 +98,9 @@ Namespaces:
 *  dns keys recover        Recover deleted key for a domain.
 *  dns list                Display records for a domain.
 *  dns update              Update record entry for a domain.
+*  dnssec create           Create DNSSEC key.
```
Comment (aegiap) [inline gandicli.man.rst]: @grigouze please add dnssec methods documentation with parameters list below in the gandicli.man.rst (check for `gandi dns update`).

Comment (grigouze) [reply gandicli.man.rst]: @aegiap done

File: gandi/cli/commands/dnssec.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+
+@cli.group(name='dnssec')
+@pass_gandi
+def dnssec(gandi):
+    """Commands related to dnssec."""
+
+
+@dnssec.command()
+@click.option('--flags', default=None, help='Flags')
```
Comment (romuald) [inline gandi/cli/commands/dnssec.py]: I think you need to specify the type here otherwise you'll get a string and the API will refuse it

Comment (grigouze) [reply gandi/cli/commands/dnssec.py]: @romuald done

File: gandi/cli/commands/dnssec.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+@cli.group(name='dnssec')
+@pass_gandi
+def dnssec(gandi):
+    """Commands related to dnssec."""
+
+
+@dnssec.command()
+@click.option('--flags', default=None, type=click.INT, help='Flags')
+@click.option('--algorithm', default=None, type=click.INT, help='Algorithm')
+@click.option('--public_key', default=None, help='Public key')
```
Comment (MoiTux) [inline gandi/cli/commands/dnssec.py]: Those 3 parameters seems mandatory by the APIv3.   
The first two are "enum", maybe the "enum" can be define here to ? ([see](https://github.com/Gandi/gandi.cli/blob/master/gandi/cli/core/params.py))
The public-key seems to be encoded in bases 64 might be worth saying it in the help message

File: gandi/cli/modules/dnssec.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        """Create a dnssec key."""
+        fqdn = fqdn.lower()
+
+        params = {
+            'flags': flags,
+            'algorithm': algorithm,
+            'public_key': public_key,
+        }
+
+        result = cls.call('domain.dnssec.create', fqdn, params)
```
Comment (MoiTux) [inline gandi/cli/modules/dnssec.py]: Shouldn't you handle the `background` option and/or do a `cls.display_progress(result)` few create/delete method already do that

Comment (grigouze) [reply gandi/cli/modules/dnssec.py]: @MoiTux fixed

**Comment by aegiap** [comment]
@grigouze please add these description of public methods in your commit:

```
diff --git a/gandicli.man.rst b/gandicli.man.rst
index 52a06e6..5099b37 100644
--- a/gandicli.man.rst
+++ b/gandicli.man.rst
@@ -246,6 +246,12 @@ Details:
 
 * ``gandi disk snapshot resource`` will create a snapshot on the fly from a disk. Possible option is ``--name TEXT`` for the name of the snapshot (if not present, will be autogenerated). The operation can be done as background process using the option ``--background`` (or ``--bg``).
 
+* ``gandi dnssec create`` will create a DNSSEC key for the domain ``domain.tld``. It taks 4 parameters, ``fqdn`` which is the domaine for which we want to create the key and ``flag`` which is the flag to use for creation (ZSK or KSK) and ``algorithm`` for the choice of the algorithm for the key and the ``public_key`` in a base64 encoded form.
+
+* ``gandi dnssec delete ressource`` will remove a DNSSEC key identified by a ressource identificator.
+
+* ``gandi dnssec list domain.tld`` will list DNSSEC keys for domain ``domain.tld``.
+
 * ``gandi docker`` will setup ssh forwarding towards a gandi VM, remotely feeding a docker unix socket. This, for example, can be used for zeroconf access to scripted temporary build VMs. The ``--vm`` option alters the ``dockervm`` configuration parameter and can be used to set the VM used for future docker connections. ``dockervm`` can also be set locally for per-project vms (See ``gandi config``). *NOTE*: passing option parameters to docker require the usage of the POSIX argument parsing ``--`` separator. *NOTE*: a local docker client is required for this command to operate.
```

**Comment by aegiap** [comment]
@grigouze also we should add information about the ``gandi dnsssec create|list|delete`` methods and ``gandi dns keys create|delete|info|recover|list``.

We could also add a ``gandi dnssec info ressource_id`` to get the flag and algo of a specific DNSSEC key.

**Comment by hal869** [comment]
Can we please get this merged, at least to master branch?

**Comment by aegiap** [comment]
@grigouze I pushed a new commit in your branch with the documentation of DNSSEC methods. Feel free to squash my commit.

**Pull request merged**